docs/customization_guide/inference_protocols.md

```diff
@@ -114,6 +114,54 @@ These options can be used to configure the KeepAlive settings:
+
+```
+--grpc-restricted-protocol=<protocol_1>,<protocol_2>,...:<restricted-header>=<restricted-key>
+```
+
+The option can be specified multiple times to specifies multiple groups of
+protocols with different restriction settings.
+
+* `protocols` : A comma-separated list of protocols to be included in this
+group. Note that currently a given protocol is not allowed to be included in
+multiple groups. The following protocols are currently recognized by all network
+protocol types mentioned above:
+
+  * `health` : Health endpoint defined for [HTTP/REST](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#health) and [GRPC](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#health-1). For GRPC endpoint, this value also exposes [GRPC health check protocol](https://github.com/triton-inference-server/common/blob/main/protobuf/health.proto).
+  * `metadata` : Server / model metadata endpoints defined for [HTTP/REST](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#server-metadata) and [GRPC](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#server-metadata-1).
+  * `inference` : Inference endpoints defined for [HTTP/REST](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#inference) and [GRPC](https://github.com/kserve/kserve/blob/master/docs/predict-api/v2/required_api.md#inference-1).
+  * `shared-memory` : [Shared-memory endpoint](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_shared_memory.md).
+  * `model-config` : [Model configuration endpoint](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_model_configuration.md).
+  * `model-repository` : [Model repository endpoint](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_model_repository.md).
+  * `statistics` : [statistics endpoint](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_statistics.md).
+  * `trace` : [trace endpoint](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_trace.md).
+  * `logging` : [logging endpoint](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_logging.md).
+
+* `restricted-header` : The GRPC request header to be checked when a request to
+the protocol is received.
+
+* `restricted-key` : The value of the header to be matched in order to preceed
+the process of the specified protocols.
+
+#### Example
+
+To start server with the two GRPC endpoints in use case described above, the following command line arguments can be set to start "standard inference" endpoint on port 8002 and "administration" endpoint with health check on port 8003:
+
+```
+tritonserver \
+--endpoint-config=infer,type=grpc --endpoint-config=infer,grpc-port=8002 --endpoint-config=infer,protocols=health,metadata,inference \
```

Comment:
this set of options is quite different from the `--grpc-restricted-protocol=` options above.

how are they reconciled?

Reply:
I missed the update of example section, so the current way is that `--grpc-restricted-protocol` will be recognized instead of `--endpoint-config`, it is going to just enforce additional check on the specified protocols (different from HTTP/GRPC, it is how we call the "endpoint APIs").

So this example will be, where only KServe protocol will not be restricted, others need to have proper header "<admin-key> : <admin-value>". And there is still only one GRPC endpoint running to accept all GRPC requests
```
tritonserver --grpc-restricted-protocol=shared-memory,model-config,model-repository,statistics,trace:<admin-key>=<admin-value>
```